### PR TITLE
make .schema to accept objects that respond to #schema

### PR DESCRIPTION
**Pull request by ttdonovan**
An attempt to resolve issue #136.

### CODE REVIEW

File: lib/dry/validation/schema.rb [hunk 1]
```diff
@@ -53,6 +53,8 @@ def self.create_class(target, other = nil, &block)
             Validation.Schema(parent: target, build: false) do
               other.schema.each { |attr, type| required(attr).filled(type) }
             end
+          elsif other.respond_to?(:schema) and other.schema.is_a?(self)
```
Comment (jodosha) [inline lib/dry/validation/schema.rb]: @ttdonovan Shouldn't `and` be replaced with `&&`?

Comment (ttdonovan) [inline lib/dry/validation/schema.rb]: Yes it should.

**Comment by jodosha** [comment]
@ttdonovan Thanks for this PR. I've tried with the hanami-validations integration and it works great. 👍

**Comment by AMHOL** [comment]
This looks good to me, just waiting for @timriley to check it out too

**Comment by timriley** [comment]
Looks good to me - I'll merge, thanks!

**Pull request merged**